### ISSUE TITLE
[qt5-base:x64-windows] build failure

### ISSUE THREAD

**Issue by cenit** [microsoft/vcpkg#12341]
**Host Environment**
- OS: Windows
- Compiler: VS 2019 16.6.3 (latest version)

**To Reproduce**
`./vcpkg upgrade --no-dry-run`

**Failure logs** 
```
cl : Command line warning D9025 : overriding '/Fo./' with '/Foqlibraryinfo_final.obj'
WARNING: Failure to find: C:/src/vcpkg/buildtrees/qt5-base/x64-windows-dbg/src/corelib/qt5cored_metatypes.json
WARNING: Failure to find: C:/src/vcpkg/buildtrees/qt5-base/x64-windows-dbg/src/corelib/qt5core_metatypes.json
Error: dependent 'C:\src\vcpkg\buildtrees\qt5-base\x64-windows-dbg\src\corelib\qt5cored_metatypes.json' does not exist.
jom: C:\src\vcpkg\buildtrees\qt5-base\x64-windows-dbg\src\corelib\Makefile [debug] Error 2
jom: C:\src\vcpkg\buildtrees\qt5-base\x64-windows-dbg\src\Makefile [sub-corelib-make_first] Error 2
jom: C:\src\vcpkg\buildtrees\qt5-base\x64-windows-dbg\Makefile [sub-src-make_first] Error 2
```

**Additional context**
Upgrading my vcpkg install folder today, with the QT5 port updated to 5.15 #11596 , lead to failure.
Re-launching the installation gives the same failure.
Other issues regarding latest qt5 update are also emerging from ci #12229 #11130

**Comment by Neumann-A** [comment]
>Other issues regarding latest qt5 update are also emerging from ci #12229 #11130

Nothing to see here. qt5-imageformats is sometimes a bit flaky and just need to be rerun and the qt5-base regression is a download failure of winflexbison.

I also get the warnings:
```
cl : Command line warning D9025 : overriding '/Fo./' with '/Foqlibraryinfo_final.obj'
WARNING: Failure to find: E:/qt/buildtrees/qt5-base/x64-windows-static-dbg/src/corelib/qt5cored_metatypes.json
WARNING: Failure to find: E:/qt/buildtrees/qt5-base/x64-windows-static-dbg/src/corelib/qt5core_metatypes.json
E:\qt\buildtrees\qt5-base\src\5.15.0-f256990b37.clean\src\corelib\kernel\qobjectdefs.h(0): Note: No relevant classes found. No output generated.
WARNING: Failure to find: E:/qt/buildtrees/qt5-base/x64-windows-static-dbg/src/gui/qt5guid_metatypes.json
WARNING: Failure to find: E:/qt/buildtrees/qt5-base/x64-windows-static-dbg/src/gui/qt5gui_metatypes.json
E:\qt\buildtrees\qt5-base\src\5.15.0-f256990b37.clean\src\network\access\qnetworkaccessdebugpipebackend_p.h(0): Note: No relevant classes found. No output generated.
WARNING: Failure to find: E:/qt/buildtrees/qt5-base/x64-windows-static-dbg/src/widgets/qt5widgetsd_metatypes.json
WARNING: Failure to find: E:/qt/buildtrees/qt5-base/x64-windows-static-dbg/src/widgets/qt5widgets_metatypes.json
```
but the build succeeds.  I also see the *.json files in the buildtree being generated. hmm...

**Comment by Neumann-A** [comment]
```
>CMake Error at /Users/vagrant/Data/installed/x64-osx/share/cmake/Qt5Core/Qt5CoreConfig.cmake:95 (message):
  Library not found: zstd
Call Stack (most recent call first):
  /Users/vagrant/Data/installed/x64-osx/share/cmake/Qt5Core/Qt5CoreConfig.cmake:285 (_qt5_Core_process_prl_file)
  /Volumes/data/work/1/s/scripts/buildsystems/vcpkg.cmake:415 (_find_package)
  /Users/vagrant/Data/installed/x64-osx/share/cmake/Qt5Gui/Qt5GuiConfig.cmake:231 (find_package)
  /Volumes/data/work/1/s/scripts/buildsystems/vcpkg.cmake:415 (_find_package)
  /Users/vagrant/Data/installed/x64-osx/share/cmake/Qt5Widgets/Qt5WidgetsConfig.cmake:231 (find_package)
  /Volumes/data/work/1/s/scripts/buildsystems/vcpkg.cmake:415 (_find_package)
  software/VO/AlternativeVO/CMakeLists.txt:7 (find_package)
```

From https://github.com/microsoft/vcpkg/pull/12229#issuecomment-656078755: Update qt5-base build-depends to include zstd. (Currently trying to find the place actually introducing zstd)

**Comment by cenit** [comment]
@JackBoosY why requires:repro?

**Comment by JackBoosY** [comment]
@cenit Because I can't reproduce this issue on my machine.

**Comment by talregev** [comment]
Also happen to me:
`vcpkg.exe install opencv3[qt,ipp,world,core,dnn,flann,jpeg,opengl,png,tiff,webp,tbb,ffmpeg,jasper,contrib,eigen] --triplet x64-windows
`

[install-x64-windows-dbg-out.log](https://github.com/microsoft/vcpkg/files/4937661/install-x64-windows-dbg-out.log)
tldr (the log):

```
C:\Users\User\Desktop\vcpkg\buildtrees\opencv3\src\3.4.7-8b40424ca7.clean\modules\cvv\src\qtutil\../util/observer_ptr.hpp(177): error C2039: 'logic_error': is not a member of 'std'
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.26.28801\include\memory(24): note: see declaration of 'std'

C:\Users\User\Desktop\vcpkg\buildtrees\opencv3\src\3.4.7-8b40424ca7.clean\modules\cvv\src\qtutil\../util/observer_ptr.hpp(174): note: while compiling class template member function 'void cvv::util::ObserverPtr<QVBoxLayout>::enforceNotNull(void) const'
C:\Users\User\Desktop\vcpkg\buildtrees\opencv3\src\3.4.7-8b40424ca7.clean\modules\cvv\src\qtutil\../util/observer_ptr.hpp(96): note: see reference to function template instantiation 'void cvv::util::ObserverPtr<QVBoxLayout>::enforceNotNull(void) const' being compiled
C:\Users\User\Desktop\vcpkg\buildtrees\opencv3\src\3.4.7-8b40424ca7.clean\modules\cvv\src\qtutil\collapsable.hpp(134): note: see reference to class template instantiation 'cvv::util::ObserverPtr<QVBoxLayout>' being compiled
C:\Users\User\Desktop\vcpkg\buildtrees\opencv3\src\3.4.7-8b40424ca7.clean\modules\cvv\src\qtutil\../util/observer_ptr.hpp(177): error C2065: 'logic_error': undeclared identifier
```

**Comment by Neumann-A** [comment]
@talregev: wrong issues? This one is about qt5-base not opencv3.

**Comment by talregev** [comment]
yes. I delete it.

**Comment by Neumann-A** [comment]
@cenit is this still an issue? If yes could you attach both logs (err and out) ?

**Comment by cenit** [comment]
yes still an issue, I can't use any qt5 project on windows via vcpkg from master anymore

[build-x64-windows-dbg-err.log](https://github.com/microsoft/vcpkg/files/4988536/build-x64-windows-dbg-err.log)
[build-x64-windows-dbg-out.log](https://github.com/microsoft/vcpkg/files/4988537/build-x64-windows-dbg-out.log)
[config-x64-windows-dbg-err.log](https://github.com/microsoft/vcpkg/files/4988538/config-x64-windows-dbg-err.log)
[config-x64-windows-dbg-out.log](https://github.com/microsoft/vcpkg/files/4988539/config-x64-windows-dbg-out.log)
[config-x64-windows-rel-err.log](https://github.com/microsoft/vcpkg/files/4988540/config-x64-windows-rel-err.log)
[config-x64-windows-rel-out.log](https://github.com/microsoft/vcpkg/files/4988541/config-x64-windows-rel-out.log)
[stdout-x64-windows.log](https://github.com/microsoft/vcpkg/files/4988542/stdout-x64-windows.log)

**Comment by Neumann-A** [comment]
Have you tried without the symlink?

I see
`IC:\Users\ssinigar\Codice\vcpkg_openmvg\buildtrees\qt5-base\src\5.15.0-f256990b37.clean\include\QtCore`
And
`IC:\src\vcpkg_openmvg\buildtrees\qt5-base\src\5.15.0-f256990b37.clean/include\QtCore`

The first path is also very long.

**Comment by cenit** [comment]
I launched the build from C:\src, which is a subst of C:\Users\ssinigar\Codice, I don't know how it also expanded it to the full path. Also, no problem with any other port which has the infamous maximum path length
This also was a supported way, and it's very difficult also to propagate any change to it

**Comment by Neumann-A** [comment]
How did you subst that? I always get a "invalid paramter error C:\src" if i try to `subst C:\src <pathtosubst>`

**Comment by Neumann-A** [comment]
after testing your `mklink` is the reason it fails. If you directly use the base path it works. If you truly use a `subst` it also works.  If you use `mklink` as a junction `/J` instead of a normal symlink it also works. So the symbolic link is the issue.

**Comment by cenit** [comment]
Let me investigate my own setup then. Thanks for taking time to already test many configs

**Comment by Neumann-A** [comment]
@cenit: I assume you found a solution? (switching to `nmake` can also solve the problem but I only tested that manually outside the portfile which than run into other problems since the env was not correctly setup.)

**Comment by cenit** [comment]
Yes it was a symlink and not a junction. Strangely enough it fails mixing the two paths. I was able to complete the build from the original (even if longer) folder

**Comment by introspection3** [comment]
the problem is still here ,please open it @cenit

**Comment by introspection3** [comment]
jom 1.1.3 - empower your cores

WARNING: Failure to find: qt5qml_metatypes.json
WARNING: Failure to find: G:/src/vcpkg/buildtrees/qt5-declarative/src/5.15.2-30e50540ea.clean/src/3rdparty/masm/assembler/*.h
WARNING: Failure to find: G:/src/vcpkg/buildtrees/qt5-declarative/src/5.15.2-30e50540ea.clean/src/3rdparty/masm/wtf/*.h
WARNING: Failure to find: G:/src/vcpkg/buildtrees/qt5-declarative/x86-windows-static-rel/src/qml/qt5qml_metatypes.json
WARNING: Failure to find: qt5qmld_metatypes.json
WARNING: Failure to find: G:/src/vcpkg/buildtrees/qt5-declarative/src/5.15.2-30e50540ea.clean/src/3rdparty/masm/assembler/*.h
WARNING: Failure to find: G:/src/vcpkg/buildtrees/qt5-declarative/src/5.15.2-30e50540ea.clean/src/3rdparty/masm/wtf/*.h
WARNING: Failure to find: G:/src/vcpkg/buildtrees/qt5-declarative/x86-windows-static-rel/src/qml/qt5qmld_metatypes.json
Cannot open foreign types file G:/src/vcpkg/installed/x86-windows-static/lib/metatypes/qt5network_metatypes.json
WARNING: Failure to find: qt5qmlmodels_metatypes.json
WARNING: Failure to find: G:/src/vcpkg/buildtrees/qt5-declarative/x86-windows-static-rel/src/qmlmodels/qt5qmlmodels_metatypes.json
WARNING: Failure to find: qt5qmlmodelsd_metatypes.json
WARNING: Failure to find: G:/src/vcpkg/buildtrees/qt5-declarative/x86-windows-static-rel/src/qmlmodels/qt5qmlmodelsd_metatypes.json
Cannot open foreign types file G:/src/vcpkg/installed/x86-windows-static/lib/metatypes/qt5network_metatypes.json
WARNING: Failure to find: qt5qmlworkerscript_metatypes.json
WARNING: Failure to find: G:/src/vcpkg/buildtrees/qt5-declarative/x86-windows-static-rel/src/qmlworkerscript/qt5qmlworkerscript_metatypes.json
WARNING: Failure to find: qt5qmlworkerscriptd_metatypes.json
WARNING: Failure to find: G:/src/vcpkg/buildtrees/qt5-declarative/x86-windows-static-rel/src/qmlworkerscript/qt5qmlworkerscriptd_metatypes.json
Cannot open foreign types file G:/src/vcpkg/installed/x86-windows-static/lib/metatypes/qt5network_metatypes.json
WARNING: Failure to find: qt5quick_metatypes.json
WARNING: Failure to find: G:/src/vcpkg/buildtrees/qt5-declarative/x86-windows-static-rel/src/quick/qt5quick_metatypes.json
WARNING: Failure to find: qt5quickd_metatypes.json
WARNING: Failure to find: G:/src/vcpkg/buildtrees/qt5-declarative/x86-windows-static-rel/src/quick/qt5quickd_metatypes.json
G:\src\vcpkg\buildtrees\qt5-declarative\src\5.15.2-30e50540ea.clean\src\quick\scenegraph\qsgrenderloop.cpp(0): Note: No relevant classes found. No output generated.
Cannot open foreign types file G:/src/vcpkg/installed/x86-windows-static/lib/metatypes/qt5network_metatypes.json
WARNING: Failure to find: qt5quickshapes_metatypes.json
WARNING: Failure to find: G:/src/vcpkg/buildtrees/qt5-declarative/x86-windows-static-rel/src/quickshapes/qt5quickshapes_metatypes.json
WARNING: Failure to find: qt5quickshapesd_metatypes.json
WARNING: Failure to find: G:/src/vcpkg/buildtrees/qt5-declarative/x86-windows-static-rel/src/quickshapes/qt5quickshapesd_metatypes.json
Cannot open foreign types file G:/src/vcpkg/installed/x86-windows-static/lib/metatypes/qt5network_metatypes.json
WARNING: Failure to find: qt5quicktest_metatypes.json
WARNING: Failure to find: G:/src/vcpkg/buildtrees/qt5-declarative/x86-windows-static-rel/src/qmltest/qt5quicktest_metatypes.json
WARNING: Failure to find: qt5quicktestd_metatypes.json
WARNING: Failure to find: G:/src/vcpkg/buildtrees/qt5-declarative/x86-windows-static-rel/src/qmltest/qt5quicktestd_metatypes.json
Cannot open foreign types file G:/src/vcpkg/installed/x86-windows-static/lib/metatypes/qt5test_metatypes.json
Cannot open foreign types file G:/src/vcpkg/installed/x86-windows-static/lib/metatypes/qt5network_metatypes.json
WARNING: Failure to find: labsmodelsplugin_metatypes.json
WARNING: Failure to find: qmlfolderlistmodelplugin_metatypes.json
WARNING: Failure to find: qmllocalstorageplugin_metatypes.json
WARNING: Failure to find: qmlsettingsplugin_metatypes.json
WARNING: Failure to find: qtqmlstatemachine_metatypes.json
WARNING: Failure to find: qmlsettingsplugind_metatypes.json
WARNING: Failure to find: qmlfolderlistmodelplugind_metatypes.json
WARNING: Failure to find: labsmodelsplugind_metatypes.json
WARNING: Failure to find: qmllocalstorageplugind_metatypes.json
WARNING: Failure to find: qtqmlstatemachined_metatypes.json
WARNING: Failure to find: labsanimationplugin_metatypes.json
WARNING: Failure to find: labsanimationplugind_metatypes.json
Cannot open foreign types file G:/src/vcpkg/installed/x86-windows-static/lib/metatypes/qt5network_metatypes.json
WARNING: Failure to find: qquicklayoutsplugin_metatypes.json
WARNING: Failure to find: qquicklayoutsplugind_metatypes.json
Cannot open foreign types file G:/src/vcpkg/installed/x86-windows-static/lib/metatypes/qt5network_metatypes.json
Cannot open foreign types file G:/src/vcpkg/installed/x86-windows-static/lib/metatypes/qt5sql_metatypes.json
Cannot open foreign types file G:/src/vcpkg/installed/x86-windows-static/lib/metatypes/qt5network_metatypes.json
Cannot open foreign types file G:/src/vcpkg/installed/x86-windows-static/lib/metatypes/qt5network_metatypes.json
WARNING: Failure to find: windowplugin_metatypes.json
WARNING: Failure to find: windowplugind_metatypes.json
WARNING: Failure to find: qmlwavefrontmeshplugin_metatypes.json
WARNING: Failure to find: qmlwavefrontmeshplugind_metatypes.json
WARNING: Failure to find: qmltestplugin_metatypes.json
WARNING: Failure to find: qmltestplugind_metatypes.json
Cannot open foreign types file G:/src/vcpkg/installed/x86-windows-static/lib/metatypes/qt5network_metatypes.json
WARNING: Failure to find: sharedimageplugin_metatypes.json
WARNING: Failure to find: sharedimageplugind_metatypes.json
Cannot open foreign types file G:/src/vcpkg/installed/x86-windows-static/lib/metatypes/qt5network_metatypes.json
Cannot open foreign types file G:/src/vcpkg/installed/x86-windows-static/lib/metatypes/qt5network_metatypes.json
Cannot open foreign types file G:/src/vcpkg/installed/x86-windows-static/lib/metatypes/qt5network_metatypes.json
Cannot open foreign types file G:/src/vcpkg/installed/x86-windows-static/lib/metatypes/qt5network_metatypes.json
Cannot open foreign types file G:/src/vcpkg/installed/x86-windows-static/lib/metatypes/qt5network_metatypes.json
Cannot open foreign types file G:/src/vcpkg/installed/x86-windows-static/lib/metatypes/qt5network_metatypes.json
Cannot open foreign types file G:/src/vcpkg/installed/x86-windows-static/lib/metatypes/qt5test_metatypes.json
jom: G:\src\vcpkg\buildtrees\qt5-declarative\x86-windows-static-rel\tools\qmlpreview\Makefile.Release [.obj\release\moc_qmlpreviewapplication.obj] Error 1
jom: G:\src\vcpkg\buildtrees\qt5-declarative\x86-windows-static-rel\tools\qmlpreview\Makefile.Release [.obj\release\moc_qmlpreviewfilesystemwatcher.obj] Error 1
jom: G:\src\vcpkg\buildtrees\qt5-declarative\x86-windows-static-rel\tools\qmlpreview\Makefile [release] Error 2
jom: G:\src\vcpkg\buildtrees\qt5-declarative\x86-windows-static-rel\tools\qmlmin\Makefile.Release [..\..\bin\qmlmin.exe] Error 1180
jom: G:\src\vcpkg\buildtrees\qt5-declarative\x86-windows-static-rel\tools\qmlmin\Makefile [release] Error 2
jom: G:\src\vcpkg\buildtrees\qt5-declarative\x86-windows-static-rel\tools\qmlprofiler\Makefile.Release [.obj\release\commandlistener.obj] Error 1
jom: G:\src\vcpkg\buildtrees\qt5-declarative\x86-windows-static-rel\tools\qmlprofiler\Makefile.Release [.obj\release\qmlprofilerdata.obj] Error 1
jom: G:\src\vcpkg\buildtrees\qt5-declarative\x86-windows-static-rel\tools\qmlprofiler\Makefile.Release [.obj\release\qmlprofilerclient.obj] Error 1
jom: G:\src\vcpkg\buildtrees\qt5-declarative\x86-windows-static-rel\tools\qmlprofiler\Makefile [release] Error 2
jom: G:\src\vcpkg\buildtrees\qt5-declarative\x86-windows-static-rel\tools\qmllint\Makefile.Release [..\..\bin\qmllint.exe] Error 1180
jom: G:\src\vcpkg\buildtrees\qt5-declarative\x86-windows-static-rel\tools\qmllint\Makefile [release] Error 2
jom: G:\src\vcpkg\buildtrees\qt5-declarative\x86-windows-static-rel\tools\Makefile [sub-qmllint-make_first] Error 2
jom: G:\src\vcpkg\buildtrees\qt5-declarative\x86-windows-static-rel\tools\qmlcachegen\Makefile.Release [..\..\bin\qmlcachegen.exe] Error 1106
jom: G:\src\vcpkg\buildtrees\qt5-declarative\x86-windows-static-rel\tools\qmlcachegen\Makefile [release] Error 2
jom: G:\src\vcpkg\buildtrees\qt5-declarative\x86-windows-static-rel\tools\Makefile [sub-qmlcachegen-make_first] Error 2
jom: G:\src\vcpkg\buildtrees\qt5-declarative\x86-windows-static-rel\tools\qmltestrunner\Makefile.Release [..\..\bin\qmltestrunner.exe] Error 1180
jom: G:\src\vcpkg\buildtrees\qt5-declarative\x86-windows-static-rel\tools\qmltestrunner\Makefile [release] Error 2
jom: G:\src\vcpkg\buildtrees\qt5-declarative\x86-windows-static-rel\tools\Makefile [sub-qmltestrunner-make_first] Error 2
jom: G:\src\vcpkg\buildtrees\qt5-declarative\x86-windows-static-rel\tools\Makefile [sub-qmlmin-make_first] Error 2
jom: G:\src\vcpkg\buildtrees\qt5-declarative\x86-windows-static-rel\tools\Makefile [sub-qmlprofiler-make_first] Error 2
jom: G:\src\vcpkg\buildtrees\qt5-declarative\x86-windows-static-rel\tools\Makefile [sub-qmlpreview-make_first] Error 2
jom: G:\src\vcpkg\buildtrees\qt5-declarative\x86-windows-static-rel\Makefile [sub-tools-make_first] Error 2

**Comment by JackBoosY** [comment]
@introspection3 This is an unofficial issue, please open a new issue to report that.
